### ISSUE TITLE
Require react-relative-portal makes my code crash

### ISSUE THREAD

**Issue by stormcleric** [smartprogress/react-relative-portal#10]
Hi there,

I'm having a problem using react-relative-portal. It's kinda weird tbh.

I've got a component written in React and I want to use a Relative Portal in it.

Just by doing var RelativePortal = require('react-relative-portal'); the generated js breaks my app:

Uncaught ReferenceError: Components is not defined

How can I avoid this issue?

**Comment by sunify** [comment]
Can you provide your build config?

**Comment by stormcleric** [comment]
I'm building through webpack:

```
/// <binding ProjectOpened='Watch - Development' />
"use strict";
var webpack = require('webpack');
var ExtractTextPlugin = require('extract-text-webpack-plugin');
var WebpackNotifierPlugin = require('webpack-notifier');
var moment = require('moment');
var CommonsChunkPlugin = require("webpack/lib/optimize/CommonsChunkPlugin");
moment.locale('es');

var path = require('path');

const PATHS = {
    client: path.join(__dirname, 'Scripts', 'app', 'index.client.js'),   
    build: path.join(__dirname, 'Scripts', 'build')
};

console.log(PATHS.app);

module.exports = {
    entry: {        
        client: PATHS.client        
    },
    output: {
        path: PATHS.build,
        filename: '[name].js'
    },
    resolve: {
        extensions: ['', '.ts', '.js', '.jsx']
    },
    devServer: {
       
        host: "localhost",
        port: 9000,
        contentBase: [path.join(__dirname, 'Scripts', 'build')]
       
    },
    module: {
        // you can tell webpack to avoid parsing for dependencies in any files matching an Array of regex patterns
        noParse: [
			/(node_modules|~)\/(crappy\-bundled\-lib|jquery)\//gi
        ],

        //preLoaders: [
		//	// before hitting the actual loaders, load any sourcemaps specified by npm modules
		//	{ loader: 'source-map' }
        //],
        loaders: [
            {
                test: /\.jsx?$/,
                exclude: /node_modules/,
                loaders: ['babel?presets[]=react,compact=false']
                /*,query:
                {
                    // Esto se usa con babel-loader http://stackoverflow.com/questions/33460420/babel-loader-jsx-syntaxerror-unexpected-token
                    presets: ['react'],
                    compact: false
                },*/
            },
            // bundle LESS and CSS into a single CSS file, auto-generating -vendor-prefixes
            {
                test: /\.(less|css)$/,
                exclude: /\b(some\-css\-framework|whatever)\b/i,
                loader: ExtractTextPlugin.extract("style-loader", "css-loader!less-loader")
                //test: /\.css$/,
                //loader: ExtractTextPlugin.extract('style', 'css')
            },            
            {
                test: /\.(jpe|jpg|woff|woff2|eot|ttf|svg|png|ico|gif)(\?.*$|$)/,
                loader: 'file'
            },
            {
                test: /\.js$/,
                exclude: ['/node_modules/','/scripts/'],
                loader: 'babel?compact=false',
                query: { compact: false }
            }            
        ]
    },
    externals: {
        // Use external version of React (from CDN for client-side, or bundled with ReactJS.NET for server-side)
        // Comment this out if you want to load your own version of React
        react: 'React'
    },
    plugins: [
      new WebpackNotifierPlugin(),
      new ExtractTextPlugin('style.min.css', { allChunks: true })
      //,new webpack.HotModuleReplacementPlugin()
    
      
    ],
    devtool: 'source-map'
};
```

**Comment by sunify** [comment]
And minimal example that crashes with this config and react-relative-portal require

**Comment by sunify** [comment]
I am sorry, but i have no time to read this abyss of code, please extract minimal example with this issue.

**Comment by stormcleric** [comment]
I've updated my previous comment - there was no need of that much overhead. The problem is simple: if I add RelativePortal to react-select-plus it works fine with the provided examples. But if I then import the new react-select-plus into my project, it complains about the code. Please note that react-select-plus works fine in my project when I remove the react-relative-portal code.

**Comment by stormcleric** [comment]
I'm investigating further on this issue. It has nothing to do with react-select-plus, seems that the problem arises when using react-relative-portal in my webpack setup.

**Comment by stormcleric** [comment]
Ok, I've found it. It seems react-relative-portal won't work on isomorphic setups because of this line:

`document.body.addEventListener('mousewheel', (0, _debounce2.default)(fireListeners, 100, true));`

document.body is either null or undefined so that statement fails.

Right now I came to this solution:

`if (document != null && document != undefined && document.body != null && document.body != undefined){  
    document.body.addEventListener('mousewheel', (0, _debounce2.default)(fireListeners, 100, true));
  }
`

Yeah, not much elegant, but avoids the error. @sunify, mind to tell me what should I do to test the portal works ok? I guess I'll just need to test the mousewheel but I don't know in which way will that line affect other parts of the compo.
